wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/launcher/TerminalAgentLauncherImpl.java

```diff
@@ -28,10 +32,21 @@
 public class TerminalAgentLauncherImpl extends AbstractAgentLauncher {
     protected static final Logger LOG = LoggerFactory.getLogger(TerminalAgentLauncherImpl.class);
 
+    private final String runCommand;
+
     @Inject
     public TerminalAgentLauncherImpl(@Named("che.agent.dev.max_start_time_ms") long agentMaxStartTimeMs,
-                                     @Named("che.agent.dev.ping_delay_ms") long agentPingDelayMs) {
+                                     @Named("che.agent.dev.ping_delay_ms") long agentPingDelayMs,
+                                     @Named("machine.terminal_agent.run_command") String runCommand) {
```

Comment:
I don't think so, it may be confusing for people, as long as executable is named `che-websocket-terminal` and agent name is terminal, the command should be named appropriately.